app/src/main/contrib/components/Responsibilities/BurdenEstimates/UploadBurdenEstimatesForm.tsx

```diff
@@ -157,14 +156,20 @@ export class UploadEstimatesFormComponent extends React.Component<UploadEstimate
             this.setState({
                 uploadErrors: uploadErrors,
                 isUploading: false,
-                progress: 0
+                progress: 0,
+                file: null
             });
         });
     };
 
     componentDidUpdate(prevProps: Readonly<UploadEstimatesProps>) {
-        if (prevProps.url == null && this.props.url != null) {
-            this.uploadClient.upload();
+        if (prevProps.url != this.props.url && this.props.url != null) {
+            if (this.uploadClient.files[0].progress(false) == 1){
+                this.uploadClient.files[0].retry();
```

Comment:
This was the cause of the weird behaviour I was seeing. If the file has already been uploaded, and the user just hits "upload" again without changing the file (e.g. because they've changed the metadata), the client doesn't reupload it because it thinks it's already done that successfully. In reality the url will have changed (the client doesn't seem to detect this and invalidate progress) so it needs to re-upload it to thew new url.